rocAL/rocAL/source/randombboxcrop_reader.cpp

```diff
@@ -45,9 +45,35 @@ void RandomBBoxCropReader::init(const RandomBBoxCrop_MetaDataConfig &cfg)
     _seed = cfg.seed();
 }
 
-void RandomBBoxCropReader::set_meta_data(std::shared_ptr<MetaDataReader> meta_data_reader)
+
+void RandomBBoxCropReader::set_meta_data(std::shared_ptr<MetaDataReader> meta_data_reader,MetaDataReaderType type)
 {
-    _meta_data_reader = std::static_pointer_cast<COCOMetaDataReader>(meta_data_reader);
+
+    switch(type) {
```

Comment:
it shouldn't check the reader type here. Instead access the function through base class

Reply:
Made changes, testing it with SSD training

Reply:
Tested, SSD Training converged